pkg/ottl/ottlfuncs/README.md

```diff
@@ -271,7 +272,7 @@ The `replace_all_patterns` function replaces any segments in a string value or k
 
 If one or more sections of `target` match `regex` they will get replaced with `replacement`.
 
-The `replacement` string can refer to matched groups using [regexp.Expand syntax](https://pkg.go.dev/regexp#Regexp.Expand).
+The `replacement` string can refer to matched groups using [regexp.Expand syntax](https://pkg.go.dev/regexp#Regexp.Expand). `replacementFormat` is an optional string argument that specifies the format of the replacement.
```

Comment:
I've described it in the example below:
`replace_pattern(name, "^kube_([0-9A-Za-z]+_)", "$$1.", SHA256, "k8s.%s")`

Reply:
I think that's a good example, but it wouldn't be immediately obvious to me (especially if I don't have programming experience) how far I can deviate from that example. I might wonder if I have `%s.k8s.%s` whether it would that just substitute the function output for `%s`, or if there are other letters I could use.

Reply:
Added more detail on this, that hopefully makes it more clear.